### DISCUSSION TITLE
Problem with smooth graphics

### DISCUSSION THREAD

**Discussion by DamirAS2020** [Bodmer/TFT_eSPI#2523] [General]
#include <TFT_eSPI.h>       // Include the graphics library
TFT_eSPI tft = TFT_eSPI();  // Create object "tft"
uint8_t radius = 110;
uint8_t len = 95;
uint16_t centreX = 120, centreY = 120;

void setup() {

  tft.init();
  tft.setRotation(1);
  tft.fillScreen(TFT_BLACK);
  pinMode(26, OUTPUT);  // lcd light
  digitalWrite(26, LOW);
  digitalWrite(26, HIGH);

  for (float angle = 0; angle < 360; angle += 3.75) {
    if (int(angle) != angle)
      tft.drawArc(centreX, centreY, radius, len, angle + 0.75, angle - 0.75 + 3.75, TFT_WHITE, TFT_BLACK, 1);
    else
      tft.drawArc(centreX, centreY, radius, len, angle + 0.75, angle - 0.75 + 3.75, TFT_WHITE, TFT_BLACK, 1);
  }
}

void loop() {}
![IMG_8164](https://user-images.githubusercontent.com/76564555/226794897-d2efb951-9703-4136-8f05-2f689ce231c9.JPG)

#include <TFT_eSPI.h>       // Include the graphics library
TFT_eSPI tft = TFT_eSPI();  // Create object "tft"
uint8_t radius = 110;
uint8_t len = 95;
uint16_t centreX = 120, centreY = 120;

void setup() {

  tft.init();
  tft.setRotation(1);
  tft.fillScreen(TFT_BLACK);
  pinMode(26, OUTPUT);  // lcd light
  digitalWrite(26, LOW);
  digitalWrite(26, HIGH);

  for (float angle = 0; angle < 360; angle += 3.75) {
    if (int(angle) != angle)
      tft.drawArc(centreX, centreY, radius, len, angle + 0.75, angle - 0.75 + 3.75, TFT_WHITE, TFT_BLACK, 1);
 //   else
 //     tft.drawArc(centreX, centreY, radius, len, angle + 0.75, angle - 0.75 + 3.75, TFT_WHITE, TFT_BLACK, 1);
  }
}

After comment 2 lines (else and next) graphics become more or less smooth

![IMG_8162](https://user-images.githubusercontent.com/76564555/226795317-3a00f124-f012-46b8-8a81-0a71e4db05ec.JPG)

What is a problem?

ESP32-WROOM-ESP32  Arduino IDE 2.0.4   ILI9341 2.2" 240x320

Thanks,

Damir Salakhov

**Comment by Bodmer** [comment]
The functions accept integer angles 0-360 so floating point values will be rounded down.

**Reply by DamirAS2020** [reply]
I tried to change cpp and h files both drawArc and drawSmoothArc.
Result looks good.
Thanks.
![IMG_8169](https://user-images.githubusercontent.com/76564555/226905144-a4897402-96bc-47bd-bc78-6bd1b3534add.JPG)
